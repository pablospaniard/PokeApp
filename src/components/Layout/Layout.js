import React, {Component, Fragment} from 'react'
import PropTypes from 'prop-types'
import {Button, Grid, LinearProgress} from '@material-ui/core'
import styled from 'styled-components'

import PokemonsListContainer from '../../containers/PokemonListContainer/PokemonListContainer'

const StyledGrid = styled(Grid)`
  text-align: center;
`

class Layout extends Component {
  state = {
    pokemons: {},
    step: 0
  }

  static getDerivedStateFromProps(nextProps) {
    return {
      pokemons: nextProps.pokemons.results
    }
  }

  onGetPokemonsHandler = () => {
    this.props.getPokemons()
    this.setState({
      step: 1
    })
  }

  onDeletePokemonsHandler = () => {
    this.props.deletePokemons()
    this.setState({
      step: 0
    })
  }

  render() {
    const {pokemons, step} = this.state
    const {loading} = this.props
    let content
    const buttons = (
      <Grid
        item
        xs={12}
        style={{display: 'flex', justifyContent: 'space-between'}}
      >
        <Button
          variant="outlined"
          color="primary"
          onClick={this.onGetPokemonsHandler}
        >
          Get Pokemons
        </Button>
        <Button
          variant="outlined"
          color="primary"
          onClick={this.onDeletePokemonsHandler}
        >
          Delete Pokemons
        </Button>
      </Grid>
    )
    switch (step) {
      case 0:
        content = <Fragment>{buttons}</Fragment>
        break
      case 1:
        content = (
          <Fragment>
            {buttons}
            <Grid item xs={12}>
              {!loading ? (
                <PokemonsListContainer pokemons={pokemons} />
              ) : (
                <LinearProgress />
              )}
            </Grid>
          </Fragment>
        )
        break
      default:
        content = <Fragment>{buttons}</Fragment>
    }
    return (
      <StyledGrid
        container
        className="App"
        justify="center"
        direction="column"
        style={{width: '25%', margin: 'auto'}}
      >
        {content}
      </StyledGrid>
    )
  }
}

Layout.propTypes = {
  getPokemons: PropTypes.func,
  deletePokemons: PropTypes.func,
  pokemons: PropTypes.object,
  error: PropTypes.object,
  loading: PropTypes.bool
}

export default Layout
