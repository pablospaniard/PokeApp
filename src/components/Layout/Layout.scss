@import '../../scss/vars.scss';
.Container {
  font-size: 16px;
}
