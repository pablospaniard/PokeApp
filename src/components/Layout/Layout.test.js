import React from 'react'
import {shallow} from 'enzyme'
import Layout from './Layout'
import {LinearProgress} from '@material-ui/core'

describe('Layout component should work as expected', () => {
  const props = {
    pokemons: {},
    err: null,
    loading: false
  }

  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Layout {...props} />)
  })

  it('match snapshot', () => {
    expect(wrapper).toMatchSnapshot('Layout_snapshot_1')
  })

  it('should show Spinner while loading', async () => {
    wrapper.setState({step: 1})
    wrapper.setProps({loading: true})
    expect(wrapper.find(LinearProgress).length).toEqual(1)
  })
})
