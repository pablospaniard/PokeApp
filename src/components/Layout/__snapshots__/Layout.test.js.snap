// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout_snapshot_1 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Layout
    err={null}
    loading={false}
    pokemons={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Unknown>
        <WithStyles(Grid)
                item={true}
                style={
                        Object {
                                "display": "flex",
                                "justifyContent": "space-between",
                              }
                }
                xs={12}
        >
                <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
                        variant="outlined"
                >
                        Get Pokemons
                </WithStyles(Button)>
                <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
                        variant="outlined"
                >
                        Delete Pokemons
                </WithStyles(Button)>
        </WithStyles(Grid)>
</Unknown>,
      "className": "App",
      "container": true,
      "direction": "column",
      "justify": "center",
      "style": Object {
        "margin": "auto",
        "width": "25%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <WithStyles(Grid)
          item={true}
          style={
                    Object {
                              "display": "flex",
                              "justifyContent": "space-between",
                            }
          }
          xs={12}
>
          <WithStyles(Button)
                    color="primary"
                    onClick={[Function]}
                    variant="outlined"
          >
                    Get Pokemons
          </WithStyles(Button)>
          <WithStyles(Button)
                    color="primary"
                    onClick={[Function]}
                    variant="outlined"
          >
                    Delete Pokemons
          </WithStyles(Button)>
</WithStyles(Grid)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
              variant="outlined"
>
              Get Pokemons
</WithStyles(Button)>,
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
              variant="outlined"
>
              Delete Pokemons
</WithStyles(Button)>,
          ],
          "item": true,
          "style": Object {
            "display": "flex",
            "justifyContent": "space-between",
          },
          "xs": 12,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Get Pokemons",
              "color": "primary",
              "onClick": [Function],
              "variant": "outlined",
            },
            "ref": null,
            "rendered": "Get Pokemons",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Delete Pokemons",
              "color": "primary",
              "onClick": [Function],
              "variant": "outlined",
            },
            "ref": null,
            "rendered": "Delete Pokemons",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Unknown>
          <WithStyles(Grid)
                    item={true}
                    style={
                              Object {
                                        "display": "flex",
                                        "justifyContent": "space-between",
                                      }
                    }
                    xs={12}
          >
                    <WithStyles(Button)
                              color="primary"
                              onClick={[Function]}
                              variant="outlined"
                    >
                              Get Pokemons
                    </WithStyles(Button)>
                    <WithStyles(Button)
                              color="primary"
                              onClick={[Function]}
                              variant="outlined"
                    >
                              Delete Pokemons
                    </WithStyles(Button)>
          </WithStyles(Grid)>
</Unknown>,
        "className": "App",
        "container": true,
        "direction": "column",
        "justify": "center",
        "style": Object {
          "margin": "auto",
          "width": "25%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <WithStyles(Grid)
            item={true}
            style={
                        Object {
                                    "display": "flex",
                                    "justifyContent": "space-between",
                                  }
            }
            xs={12}
>
            <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
                        variant="outlined"
            >
                        Get Pokemons
            </WithStyles(Button)>
            <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
                        variant="outlined"
            >
                        Delete Pokemons
            </WithStyles(Button)>
</WithStyles(Grid)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
                variant="outlined"
>
                Get Pokemons
</WithStyles(Button)>,
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
                variant="outlined"
>
                Delete Pokemons
</WithStyles(Button)>,
            ],
            "item": true,
            "style": Object {
              "display": "flex",
              "justifyContent": "space-between",
            },
            "xs": 12,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Get Pokemons",
                "color": "primary",
                "onClick": [Function],
                "variant": "outlined",
              },
              "ref": null,
              "rendered": "Get Pokemons",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Delete Pokemons",
                "color": "primary",
                "onClick": [Function],
                "variant": "outlined",
              },
              "ref": null,
              "rendered": "Delete Pokemons",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": Symbol(react.fragment),
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
